.travis.yml

```diff
@@ -1,188 +1,109 @@
-        - export JAVA_HOME=~/jdk-11.0.5+10
-        - export PATH=${JAVA_HOME}/bin:$PATH
-        - java --version
-      install: skip
-      script:
-        - ./.travis/setup_database
-        - ./.travis/run_smoke_tests
-    - stage: deploy artifacts
-      env: "Upload to Github Releases"
+      script: [ $VERIFY_STAGE/smoke-testing/run || $REPORT_FAIL ]
+
+    - stage: Verify
+      name: "Analysis: PMD Verification"
+      script: [ $VERIFY_STAGE/pmd-verification/run || $REPORT_FAIL ]
+    
+    - stage: Verify
+      name: "Analysis: Checkstyle"
+      script: [ $VERIFY_STAGE/checkstyle/run || $REPORT_FAIL ]
+    
+    - stage: Verify
+      name: "Analysis: Java Formatting"
+      script: [ $VERIFY_STAGE/java-formatting/run || $REPORT_FAIL ]
+    
+    - stage: Verify
+      name: "Analysis: Check Links and YAML Syntax"
+      script: [ $VERIFY_STAGE/check-links-and-yaml-syntax/run || $REPORT_FAIL ]
+      
+    - stage: Verify
+      name: "Analysis: Code Convention Checks"
+      script: [ $VERIFY_STAGE/code-convention-checks/run || $REPORT_FAIL ]
+
+    - stage: Verify
+      name: "Analysis: Find Unused or Unmatched Tests"
+      script: [ $VERIFY_STAGE/find-unused-or-unmatched-tests/run || $REPORT_FAIL ]
+
+    ########################################################################
+
+    - stage: Deploy
+      name: "Push Map Descriptions to Website"
+      if: (branch = master) and (repo = 'triplea-game/triplea') and (type != 'pull_request')
+      addons: { apt: { packages: [ python3, python3-yaml ] } }
+      script: [ $DEPLOY_STAGE/update-maps-on-website/run || $REPORT_FAIL ]     
+
+    - stage: Deploy
+      name: "Ansible Deploy to Prerelease"
+      if: (branch = master) and (repo = 'triplea-game/triplea') and (type != 'pull_request')
+      language: python
+      python: "3.6"
+      addons: { apt: { packages: [ sshpass ] } }
+      install: [ pip install ansible ]
+      script: [ $DEPLOY_STAGE/ansible-deploy-to-prerelease/run || $REPORT_FAIL ]
+
+    - stage: Deploy
+      name: "Build Installer Downloads & Upload to Github Releases"
       if: (branch = master) and (repo = 'triplea-game/triplea') and (type != 'pull_request')
-      install: skip
-      cache:
-        directories:
-          - '$HOME/.gradle'
-          - '.gradle'
-          - '~/jdk-11.0.5+10'
-      before_install:
-        - ./.travis/install_java_11.0.5
-        - export JAVA_HOME=~/jdk-11.0.5+10
-        - export PATH=${JAVA_HOME}/bin:$PATH
-        - java --version
-      script:
-        - ./.travis/setup_gpg
-        ## Update product version to include build number.
-        ## EG: replace "2.0.0" to be "2.0.15555";
-        - sed -i "s/.0 *$/.$TRAVIS_BUILD_NUMBER/" game-core/src/main/resources/META-INF/triplea/product.properties
-        - ./.travis/do_release
+      script: [ $DEPLOY_STAGE/installers/run || $REPORT_FAIL ]
       deploy:
+        on: { tags: false, repo: triplea-game/triplea, branch: master }
```

Comment:
Hmm, I think this mix of the yaml 1.1 syntax and the newer 1.2 syntax is a little bot weird here and there.
This line in particular doesn't add to readability IMO, but personal preference I guess

Reply:
Are you looking at this only from a web-browser, from a terminal, or from IDE?
In general, I found the shorter, the better for most of this, the multi-line elements lengthened the config file and stood out as most things are single line now.

How do you mean by mixing of the 1.1 vs 1.2 syntax, as far as I knew, the inline map syntax has always been equivalent.

Reply:
Only the Web browser.
I get what you were trying to achieve, but I think in this particular case (also because there are not too many deployment blocks) it doesn't really add to readability.
But I could live with either way, just wanted to point it out.

Reply:
Here is a contrast:

![Screenshot from 2020-02-06 19-25-20](https://user-images.githubusercontent.com/12397753/73998438-80f6fd80-4916-11ea-9394-90f205df9acf.png)

![Screenshot from 2020-02-06 19-25-36](https://user-images.githubusercontent.com/12397753/73998439-80f6fd80-4916-11ea-817d-82c2a8c36bc6.png)

The multi-line leads to the most indentation and makes nearby indentation much more important. The values specified are somewhat noisy, but also makes the file quite a bit longer. One issue IMO we developed with the travis file is it became very long, too much to really understand easily.